src/main/java/io/curity/authenticator/netid/NetIdAccessServerSoapClient.java

```diff
@@ -28,60 +28,109 @@
+    private NetiDAccessServerSoap _netIDAccessServerSoap = null;
+    private final NetIdAccessConfig _config;
 
+    private static final Logger _logger = LoggerFactory.getLogger(NetIdAccessServerSoapClient.class);
+
+    private static final String JAXWS_PROPERTIES_SSL_SOCKET_FACTORY = "com.sun.xml.ws.transport.https.client.SSLSocketFactory";
     private static final String JAXWS_PROPERTIES_CONNECT_TIMEOUT = "com.sun.xml.ws.connect.timeout";
     private static final String JAXWS_PROPERTIES_REQUEST_TIMEOUT = "com.sun.xml.ws.request.timeout";
 
     // Generous timeouts; we only want to make sure that the requesting thread isn't consumed indefinitely.
     private static final int CONNECT_TIMEOUT = 3000;
     private static final int REQUEST_TIMEOUT = 10000;
 
-    private final NetIdAccessConfig _config;
-
-    public NetIdAccessServerSoapFactory(NetIdAccessConfig config)
+    public NetIdAccessServerSoapClient(NetIdAccessConfig configuration)
     {
-        _config = config;
+        super(configuration);
+        _config = configuration;
     }
 
-    public NetiDAccessServerSoap create()
+    private SSLSocketFactory getSSLSocketFactory(Optional<ServerTrustCryptoStore> maybeTrustStore, Optional<ClientKeyCryptoStore> maybeClientKeyStore)
     {
-        var httpClient = _config.getHttpClient();
-
-        return withPluginClassLoader(() -> {
-            NetiDAccessServer accessServer = new NetiDAccessServer();
-            NetiDAccessServerSoap proxy = accessServer.getNetiDAccessServerSoap();
-            configureWebserviceClient((BindingProvider) proxy, buildEndpointAddress(httpClient.getScheme()));
-            return proxy;
-        });
+        @Nullable TrustManager[] trustManagers = null;
+        @Nullable KeyManager[] keyManagers = null;
+
+        try
+        {
+            if (maybeTrustStore.isPresent())
+            {
+                _logger.debug("Applying ssl server-truststore from configuration.");
+
+                var trustStore = maybeTrustStore.get().getAsKeyStore();
+
+                var trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                trustManagerFactory.init(trustStore);
+                trustManagers = trustManagerFactory.getTrustManagers();
+            }
+
+            if (maybeClientKeyStore.isPresent())
+            {
+                var keyStore = maybeClientKeyStore.get();
+                _logger.debug("Applying ssl client-keystore from configuration.");
+
+                var keyManagerFactory = KeyManagerFactory.getInstance(getDefaultAlgorithm());
+                keyManagerFactory.init(keyStore.getAsKeyStore(), keyStore.getKeyStorePassword());
+                keyManagers = keyManagerFactory.getKeyManagers();
+            }
+
+            var sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(keyManagers, trustManagers, null);
```

Comment:
If truststore is not configured, this means that `trustManagers = null`. Are you sure that the default trust will be used in that case?

Reply:
The documentation for `SSLContext.init` says so:
> Either of the first two parameters may be null in which case the installed security providers will be searched for the highest priority implementation of the appropriate factory.